packages/data-stores/src/site/types.ts

```diff
@@ -43,3 +43,28 @@ export interface CreateSiteParams {
 		theme?: string;
 	};
 }
+
+export interface ExistingSiteDetails {
+	ID: number;
+	name: string;
+	description: string;
+	URL: string;
+}
+
+export interface ExistingSiteError {
+	error: string;
+	message: string;
+}
+
+export type ExistingSiteResponse = ExistingSiteDetails | ExistingSiteError;
+
+export type MetadataDispatch = {
+	invalidateResolution: (
+		selectorName: string,
+		args: Array< any >
+	) => { type: string; selectorName: string; args: any };
+};
+
+export type MetadataSelect = {
+	getIsResolving: ( selectorName: string, args: Array< any > ) => boolean | undefined;
```

Comment:
Rather than exposing this for our tests, can we use `@wordpress/core-data`'s `isResolving`? ([see](https://github.com/Automattic/wp-calypso/pull/40158/files#r398577848))

Reply:
I've switched to using `core/data` so we can remove this type, but I've kept `getIsResolving` instead of `isResolving`. `isResolving` will return false if the underlying value isn't true, which means that it doesn't distinguish between a cached and resolved resolver (`false`) and an uncached resolver that hasn't started resolving yet (`undefined`). To be able to test that we're correctly invalidating the resolver cache, the tests need to be able to tell the difference.